docs/compile.md

```diff
@@ -1,64 +1,106 @@
+## Advanced Compiler Configurations
+Clicking on Advanced Compiler Configurations will open this panel (**M. in fig. 2 below**).
+
+![](images/a-sol-comp-adv.png)
+
+There is a radio button to choose whether to configure the compiler through the interface (**N in fig 2**) or to use a JSON file for the cofiguration (**R in fig 2**). 
+
+### Solidity or YUL
+Since the Solidity version `0.5.7`, it is possible to compile `Yul` files. Please read the ([solidity documentation about Yul](https://solidity.readthedocs.io/en/latest/yul.html)) which contain some code examples.
+You can use the language dropdown (**O. in fig 2**) to switch the language. **This dropdown list is only available for versions greater than or equal to `0.5.7`.**
+
+### Select an Ethereum fork
+The "fork selection" dropdown list (**P. in fin fig 2**) allows to compile code against a specific **ethereum hard fork**.
+The `compiler default` corresponds to the default hard fork used by a specific version.
+
+To see the name of the hard fork used in the current compilation, click the "Compilation Details" button and in the `Metadata` section there will be a sub-section called **settings**.  Open up the **settings** to see the hard fork's name.
+
+### Enable optimization
+According to the the Solidity Docs, "the optimizer tries to simplify complicated expressions, which reduces both code size and execution cost, i.e., it can reduce gas needed for contract deployment as well as for external calls made to the contract."
+
+For recent versions of Solidity, it is [recommended to enable optimization](https://blog.soliditylang.org/2020/11/04/solidity-ama-1-recap/#why-do-you-think-people-are-generally-suspicious-of-the-optimizer-and-are-they-right-to-be) .
+
+To learn more about optimization, (**Q. in the fig 2**) visit the [Solidity docs on the optimizer](https://docs.soliditylang.org/en/latest/internals/optimizer.html).
+
+To the right of the **Enable optimization** checkbox is the box to input the number of Optimization runs.  The default value is 200.  
+
+You may ask — "What is the right number of runs for my contract?" And the Solidity docs say:
+
+   > If you want the initial contract deployment to be cheaper and the later function executions to be more expensive, set it to --optimize-runs=1. If you expect many transactions and do not care for higher deployment cost and output size, set --optimize-runs to a high number.
+
+To learn more about the optimization runs, visit the [Solidity docs about Optimizer options](https://docs.soliditylang.org/en/latest/using-the-compiler.html?highlight=optimize-runs#optimizer-options).
+
+### JSON file for Compiler configuration
+Selecting the radio button next to **Use configuration file** will let you set the configuration using a JSON file (**T. in fig 2**).  Clicking the file name will open it up in the Editor. 
```

Comment:
We should add that a default/sample config file will be available in each workspace which can be edited with more available options. It is also better to tell about the limitation on the fields that can be added in config file.